frontend/simple/js/state.js

```diff
@@ -188,17 +244,23 @@ const actions = {
       // Subscribe to your fellow group member's identity contracts upon joining
       // TODO right an opposite series of operations for when someone leaves a group
       if (entry instanceof Events.AcceptInvitation && entry.data.username !== state.loggedIn) {
-        let identity = await backend.lookup(entry.data.username)
+        let identity = await namespace.lookup(entry.data.username)
         await backend.subscribe(identity)
-        await dispatch('synchronize', identity)
+        await dispatch('syncContractWithServer', identity)
+      }
+      // TODO Remove this when we have a system for limiting whom we receive messages from
+      if (entry instanceof Events.PostMessage && entry.data.from) {
+        let identity = await namespace.lookup(entry.data.from)
+        if (!state.contracts[identity]) {
+          await backend.subscribe(identity)
+          await dispatch('syncContractWithServer', identity)
+        }
```

Comment:
Let's do a video chat about this, because these two `if` blocks are again wayyy too detail-oriented for this file. They don't belong here in other words. We can't handle action-specific stuff in this function. Only general operations on `HashableAction`s allowed here. So this stuff, and the stuff above it for `AcceptInvitation`, must be moved into their appropriate places within `shared/events.js` and called via line 263 below (i.e. the `commit(`${contractId}/${type}`, {data: entry.data, hash})` line)